subsys/logging/log_msg2.c

```diff
@@ -83,7 +83,7 @@ void z_impl_z_log_msg2_runtime_vcreate(uint8_t domain_id, const void *source,
 	}
 
 	if (msg && fmt) {
-		plen = cbvprintf_package(msg->data, plen, 0, fmt, ap);
+		plen = cbvprintf_package(msg->data, (size_t)plen, 0, fmt, ap);
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/blob/89408901a81e3c99608a953ee5cbfac835a5be3a/subsys/logging/log_msg2.c#L60

Instead of casting, it would be cleaner to declare `plen` as `size_t`.

Reply:
the problem is that function `cbprintf_package` is used for calculating needed size (with null buffer) and for creating actual package. It returns `int`. Negative error code or needed size if positive so typical use is following:
```
int len = cbprintf_package(NULL, 0, 0, fmt, ...);
if (len < 0) {
  /*error*/
  return;
}

void *buf = alloc(len);

int len = cbprintf_package(buf, (size_t)len, 0, 0, fmt, ...);
```

If `cbprintf_package` would be changed to use `int` instead of `size_t` then still there would be conversion in cases like:
```
uint8_t buf[128];
int len = cbprintf_package(buf, sizeof(buf), fmt, ...);
```